R/pkg/inst/worker/daemon.R

```diff
@@ -30,8 +30,42 @@ port <- as.integer(Sys.getenv("SPARKR_WORKER_PORT"))
 inputCon <- socketConnection(
     port = port, open = "rb", blocking = TRUE, timeout = connectionTimeout)
 
+selectTimeout <- function() {
+  if (is.null(parallel:::selectChildren(timeout = 0))) {
```

Comment:
This can be tested as below:

```r
p <- parallel:::mcfork()
if (inherits(p, "masterProcess")) {
  Sys.sleep(3L)
  parallel:::mcexit(0L)
}

print("TRUE - the timeout was reached")
print("parallel:::selectChildren(timeout = 0):")
print(parallel:::selectChildren(timeout = 0))
Sys.sleep(4L)

print("PID - child is exited (but not terminated)")
print("parallel:::selectChildren(timeout = 0):")
print(parallel:::selectChildren(timeout = 0))

invisible(lapply(parallel:::selectChildren(timeout = 0), function(x) { tools::pskill(x, tools::SIGUSR1) }))
Sys.sleep(1L)

print("NULL - if there are no children")
print("parallel:::selectChildren(timeout = 0):")
print(parallel:::selectChildren(timeout = 0))
```

Per the documentation - https://stat.ethz.ch/R-manual/R-devel/library/parallel/html/children.html

> selectChildren returns `TRUE` is the timeout was reached, `FALSE` if an error occurred (e.g., if the master process was interrupted) or an integer vector of process IDs with children that have data available, or` NULL` if there are no children.

I also manually built this and checked if it really wait indefinitely with printing out.

via running 

```r
df <- createDataFrame(list(list(1L, 1, "1", 0.1)), c("a", "b", "c", "d"))
collect(gapply(df, "a", function(key, x) { x }, schema(df)))
collect(gapply(df, "a", function(key, x) { x }, schema(df)))
...  # 30 times
```